### DISCUSSION TITLE
Replace object in state while keeping reactivity in component

### DISCUSSION THREAD

**Discussion by Cpousson** [vuejs/pinia#2341] [Help and Questions]
I have a complex object in state that I save to the server and the server returns a fresh copy. 

lets say it looks like this:
rule = 
{
id:number,
someinterface:interface
someinterace2:interface
etc...
}

when I return the object to pinnia and assign the results in the pinnia action (this.rule = response) I lose all v-model connectivity in my components. This makes sense to me since I am replacing the object as a whole but I was wondering if there was a quick way to re-establish the reactivity.

**Comment by posva** [comment]
this should work so you are likely extracting properties of the object with `const { thing } = object` or similar and passing that around instead of passing `object.thing` to props and composables
